File: app/src/main/java/za/org/grassroot2/view/SingleTextView.java
package za.org.grassroot2.view;

import io.reactivex.Observable;

/**
 * Created by luke on 2017/08/10.
 */
public interface SingleTextView extends FragmentView {

    Observable<CharSequence> textInputChanged();

    void setInputDefault(CharSequence defaultValue);
    void displayErrorMessage(int messageRes);

    void setInputType(int type);
    void setImeOptions(int imeOptions);

    String getInputValue();

    void focusOnInput();
}

File: app/src/main/java/za/org/grassroot2/view/SingleTextView.kt
package za.org.grassroot2.view

import io.reactivex.Observable

/**
 * Created by luke on 2017/08/10.
 */
interface SingleTextView : FragmentView {

    val inputValue: String

    fun textInputChanged(): Observable<CharSequence>

    fun setInputDefault(defaultValue: CharSequence)
    fun displayErrorMessage(messageRes: Int)

    fun setInputType(type: Int)
    fun setImeOptions(imeOptions: Int)

    fun focusOnInput()
}
